src/arrayMethodJoin.js

```diff
@@ -4,8 +4,20 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let str = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] !== null && this[i] !== undefined) {
+        str += this[i];
+      }
+
+      if (this[i] !== this[this.length - 1]) {
+        str += separator;
+      }
```

Comment:
this is not the best solution here. in your case the method won't work correctly for all inputs.

Reply:
for example `[1,2,3,4,1].join2()` will yield `12,3,4,1` which is not correct. do you see why this is the case?

Reply:
got it! fixed